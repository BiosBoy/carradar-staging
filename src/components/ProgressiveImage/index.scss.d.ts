// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'flexCenter': string;
  'progressiveImage': string;
  'progressiveImagePreloader': string;
}
export const cssExports: CssExports;
export default cssExports;
