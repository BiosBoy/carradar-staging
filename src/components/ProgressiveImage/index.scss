.flexCenter {
  display: flex;
  align-items: center;
  justify-content: center;
}

.progressiveImage {
  box-sizing: border-box;
  position: relative;
  display: block;
  margin: 0;
  padding: 0;
  width: auto;
  height: auto;
}

.progressiveImagePreloader {
  background: transparent;
  composes: progressiveImage;
  composes: flexCenter;
}
