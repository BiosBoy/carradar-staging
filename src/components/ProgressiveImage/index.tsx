import React, { memo, useState, useCallback } from 'react'
import LoadIndicator from '../LoadIndicator'

import styles from './index.scss'
import { IProps } from './interfaces'

const ProgressiveImage = (props: IProps) => {
  const { image, preloader } = props
  const { path = '', alt, customClass } = image
  const { color = 'green', type = 'circle' } = preloader || {}

  const [imageLoadStatus, setImageLoadStatus] = useState('')

  const handlerLoaded = useCallback(() => setImageLoadStatus('loaded'), [])

  const imageToRender = (
    <img
      alt={alt}
      src={path}
      className={`${styles.progressiveImage} ${customClass}`}
      onLoad={handlerLoaded}
      style={!imageLoadStatus ? { display: 'none' } : {}}
    />
  )

  const spinner = (
    <div className={customClass}>
      <LoadIndicator color={color} type={type} />
    </div>
  )

  if (!imageLoadStatus && spinner) {
    return (
      <div className={styles.progressiveImagePreloader}>
        {imageToRender}
        {spinner}
      </div>
    )
  }

  return imageToRender
}

export default memo(ProgressiveImage)
