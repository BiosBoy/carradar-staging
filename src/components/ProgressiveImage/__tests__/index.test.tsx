import React from 'react';
import { mount } from 'enzyme';

import ProgressiveImage from '../index';

describe('ProgressiveImage', () => {
  it('should return common preloader', () => {
    const Component = mount(<ProgressiveImage image={{ path: '960/localStorage.png', alt: 'test' }} />);

    expect(Component.find('div').at(0).prop('className')).toBe('progressiveImagePreloader');
    expect(Component).toMatchSnapshot();
  });
});
