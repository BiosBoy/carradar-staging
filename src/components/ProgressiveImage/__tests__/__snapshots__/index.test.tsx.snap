// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressiveImage should return common preloader 1`] = `
<Memo(ProgressiveImage)
  image={
    Object {
      "alt": "test",
      "path": "960/localStorage.png",
    }
  }
>
  <div
    className="progressiveImagePreloader"
  >
    <img
      alt="test"
      className="progressiveImage undefined"
      onLoad={[Function]}
      src="960/localStorage.png"
      style={
        Object {
          "display": "none",
        }
      }
    />
    <div>
      <Loader
        color="green"
        type="circle"
      >
        <div
          className="spinnerCircle"
        >
          <div
            className="green"
            key="0"
          />
          <div
            className="green"
            key="1"
          />
          <div
            className="green"
            key="2"
          />
          <div
            className="green"
            key="3"
          />
          <div
            className="green"
            key="4"
          />
          <div
            className="green"
            key="5"
          />
          <div
            className="green"
            key="6"
          />
          <div
            className="green"
            key="7"
          />
          <div
            className="green"
            key="8"
          />
          <div
            className="green"
            key="9"
          />
          <div
            className="green"
            key="10"
          />
          <div
            className="green"
            key="11"
          />
        </div>
      </Loader>
    </div>
  </div>
</Memo(ProgressiveImage)>
`;
